packages/utils/src/schema/toPathSchema.ts

```diff
@@ -78,52 +78,49 @@ function toPathSchemaInternal<T = any, S extends StrictRJSFSchema = RJSFSchema,
     if (Array.isArray(schemaItems)) {
       formData.forEach((element, i: number) => {
         if (schemaItems[i]) {
-          pathSchema[i] = toPathSchemaInternal<T, S, F>(
-            validator,
-            schemaItems[i] as S,
-            `${name}.${i}`,
-            rootSchema,
-            element,
-            _recurseList
-          );
+          (pathSchema as { [key in keyof PathSchema<T>]: PathSchema<T> })[i as keyof PathSchema<T>] =
```

Comment:
Does it work to update the `pathSchema` type above? Right now it is (incorrectly?) defined as `PathSchema`. Maybe it wants to be `PathSchema<T>` . Or maybe create a local variable that is cast once to the proper type? I.e.
```
// on line 74 do this?
const keyablePathSchema = pathSchema as { [key in keyof PathSchema<T>]: PathSchema<T> });
```
This avoids multiple casts later and can be commented once as to why its being done?

And then this line and all others below become:

```
keyablePathSchema[i as keyof PathSchema<T>] = ...
```